synapseclient/models/table.py

```diff
@@ -383,6 +431,186 @@ class Table(TableSynchronousProtocol, AccessControllable):
+            from synapseclient.models import Table
+
+            syn = Synapse()
+            syn.login()
+
+            table = Table(
+                name="my_table",
+                parent_id="syn1234",
+            ).get(include_columns=True)
+
+            # You may also get the table by id:
+            table = Table(
+                id="syn4567"
+            ).get(include_columns=True)
+
+            table.columns["my_old_column"].name = "my_new_column"
+
+            # Before the data is stored in synapse you'll still be able to use the old key to access the column entry
+            print(table.columns["my_old_column"])
+
+            table.store()
+
+            # After the data is stored in synapse you'll be able to use the new key to access the column entry
+            print(table.columns["my_new_column"])
+
+    Example: Create a table with a list of columns
+        A list of columns may be passed in when creating a new table. The order of the
+        columns in the list will be the order they are stored in Synapse. If the table
+        already exists and you create the Table instance in this way the columns will
+        be appended to the end of the existing columns.
+
+            from synapseclient import Synapse
+            from synapseclient.models import Column, ColumnType, Table
+
+            syn = Synapse()
+            syn.login()
+
+            columns = [
+                Column(name="my_string_column", column_type=ColumnType.STRING),
+                Column(name="my_integer_column", column_type=ColumnType.INTEGER),
+                Column(name="my_double_column", column_type=ColumnType.DOUBLE),
+                Column(name="my_boolean_column", column_type=ColumnType.BOOLEAN),
+            ]
+
+            table = Table(
+                name="my_table",
+                parent_id="syn1234",
+                columns=columns
+            )
+
+            table.store()
+
+
+    Example: Creating a table with a dictionary of columns
+        When specifying a number of columns via a dict setting the `name` attribute
+        on the `Column` object is optional. When it is not specified it will be
+        pulled from the key of the dict.
+
+            from synapseclient import Synapse
+            from synapseclient.models import Column, ColumnType, Table
+
+            syn = Synapse()
+            syn.login()
+
+            columns = {
+                "my_string_column": Column(column_type=ColumnType.STRING),
+                "my_integer_column": Column(column_type=ColumnType.INTEGER),
+                "my_double_column": Column(column_type=ColumnType.DOUBLE),
+                "my_boolean_column": Column(column_type=ColumnType.BOOLEAN),
+            }
+
+            table = Table(
+                name="my_table",
+                parent_id="syn1234",
+                columns=columns
+            )
+
+            table.store()
+
+    Example: Creating a table with an OrderedDict of columns
```

Comment:
Is there a tangible difference between creation with a `Dict` of columns vs and `OrderedDict`? Will the order of a regular `Dict` also be respected in terms of the order that columns are displayed?

Reply:
`Dict` should retain order since python 3.7+, but since it didn't before I wouldn't assume that the behavior will remain consistent forever.

OrderedDict is the guaranteed way to provide the order, `Dict` seems like it's a "It will probably respect order" situation